OrbitCore/ProcessUtils.cpp

```diff
@@ -50,33 +50,34 @@ int IsNumeric(const char* ccharptr_CharacterList)
 
 #ifndef _WIN32
 namespace  {
-std::string PrettifyProcessName(const std::string &processName) {
-  // Idea: processName contains (if not changed by the process) a
-  // null-separated list of arguments, while the first is the path to the
-  // executable as called by the user. We check wether that first string is a
-  // file path. If so, we search for the last forward slash in this string and
-  // copy from there up to the end of the whole string. So we get the
-  // executable name and the arguments.
-
-  auto resultingName = [&]() -> std::string {
-    const auto end = std::find(processName.begin(), processName.end(), '\0');
-
-    const std::string filepath{processName.begin(), end};
-
-    struct stat st{};
-    const auto result = stat(filepath.c_str(), &st);
-
-    if (result == 0) {
-        // We found a file
-        const auto begin = std::find(std::make_reverse_iterator(end),
-                                     processName.rend(), '/');
-        return std::string{begin.base(), processName.end()};
-    }
-    return processName;
-  }();
+std::string PrettifyProcessName(const std::string &processName)
+{
+    // Idea: processName contains (if not changed by the process) a
+    // null-separated list of arguments, while the first is the path to the
+    // executable as called by the user. We check wether that first string is a
+    // file path. If so, we search for the last forward slash in this string and
+    // copy from there up to the end of the whole string. So we get the
+    // executable name and the arguments.
+
+    auto resultingName = [&]() -> std::string {
```

Comment:
Why the lambda?

Reply:
The lambda is slightly improving the data flow. Due to the lambda I can return early in the "if (result == 0)"-branch.
Without a lambda I either need to have the std::replace-line twice or store the result in a temporary variable, which I have to declare before and in which I have to write from two different locations in the function.

But like I said, it's just a minor improvement in this case and I'm perfectly fine writing this without the immediately invoked lambda.